### PR TITLE
player: fix OpenMPT test case s77.it

### LINKED ISSUE #300: Envelope length discrepancy
Test case: [newtest2.zip](https://github.com/schismtracker/schismtracker/files/9013549/newtest2.zip)

I suspect there might be an issue in envelope interpolation. Compared to OpenMPT and Impulse Tracker itself, the noise hit appears to be one tick shorter. Renders from all three trackers can be found inside the module itself.

### PR DESCRIPTION
**Pull request by mrpapersonic**
Also fixes #300.

### CODE REVIEW

File: player/sndmix.c [hunk 1]
```diff
@@ -265,12 +266,22 @@ static inline void rn_process_envelope(song_voice_t *chan, int *nvol)
...
 	}
+	
+	*nvol = vol;
+}
 
-	// Panning Envelope
-	if ((chan->flags & CHN_PANENV) && (penv->pan_env.nodes)) {
-		int envpos = chan->pan_env_position;
+
+static inline void rn_process_pan_env(song_voice_t* chan, int *nvol) {
```
Comment (sagamusix) [inline player/sndmix.c]: `nvol` parameter can be removed - volume is never adjusted here.

Comment (mrpapersonic) [reply player/sndmix.c]: Thanks, removed

File: player/sndmix.c [hunk 2]
```diff
@@ -324,11 +339,24 @@ static inline void rn_process_envelope(song_voice_t *chan, int *nvol)
 			vol = 0;
 		}
 	}
-
+
```
Comment (sagamusix) [inline player/sndmix.c]:
```suggestion

```

**Comment by sagamusix** [comment]
Thanks! I think this can be merged after removing the stray newly-added whitespace.

**Pull request merged**